tapfreighter/chain_porter.go

```diff
@@ -1474,7 +1491,7 @@ func (p *ChainPorter) unlockInputs(ctx context.Context, pkg *sendPackage) {
 	// sanity-check that we have known input commitments to unlock, since
 	// that might not always be the case (for example if another party
 	// contributes inputs).
-	if pkg.SendState < SendStateStorePreBroadcast &&
+	if pkg.SendState < SendStateBroadcastComplete &&
```

Comment:
If we remove the lease on assets when we are in state `SendStateStorePreBroadcast` or `SendStateBroadcast`, we don't clean up the parcel. Which seems wrong to me. This wasn't fully the case in the old situation because we only released coins when the parcel wasn't persisted to disk. So at least upon restart, the parcel was lost, and the assets would truly be free again. In this PR, we get situations were we remove the lease on coins, but still have parcels persisted to disk, that are loaded after a restart.

Reply:
A similar problem occurs a few lines down (and wasn't introduced by this PR, but should still be handled)

https://github.com/lightninglabs/taproot-assets/blob/e0c96614f7f5b667f4890f6467f6f793dce99111/tapfreighter/chain_porter.go#L1520

In states `SendStateStorePreBroadcast` and `SendStateBroadcast` (which are the final two states that call into `unlockInputs`) we try to release the inputs of the _anchor tx_ but we this leaves us with a situation where we still have the parcel. This PR changes the behavior and at least releases the assets, but we are still stuck with a pending parcel.

Also, we should inform the user what to do if `p.cfg.Wallet.UnlockInput` fails, because now he has a potential asset burning tx in his lnd node that we couldn't cancel.

Reply:
I agree that we should cleanup the pending parcel as well.

> Also, we should inform the user what to do if `p.cfg.Wallet.UnlockInput` fails, because now he has a potential asset burning tx in his lnd node that we couldn't cancel.

I don't think we currently call `unlockInputs` in state broadcast complete or beyond. So LND won't have a burning tx.

Reply:
Perhaps, in the worst case, if we don't cancel the pending parcel then upon resumption, the asset coins would have been spent and the transaction would fail because the UTXO would have been spent.